### PR TITLE
AEA-965 cli transfer command

### LINKED ISSUE #1558: Add CLI command to transfer wealth
**Is your feature request related to a problem? Please describe.**
Currently, we don't provide transfer functionality in CLI which is useful for development.

**Describe the solution you'd like**
Add `aea transfer {identifier} {address}` command to allow for transfers from cli.

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

cli command to transfer funds to another account

## Fixes

#1558

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/utils/package_utils.py [hunk 1]
```diff
@@ -524,3 +524,14 @@ def try_get_balance(  # pylint: disable=unused-argument
         return balance
     except ValueError as e:  # pragma: no cover
         raise click.ClickException(str(e))
+
+
+def get_wallet_from_context(ctx: Context) -> Wallet:
```
Comment (DavidMinarsch) [inline aea/cli/utils/package_utils.py:529]: Missing docstring

File: aea/cli/transfer.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+from aea.crypto.wallet import Wallet
+DEFAULT_SETTLE_TIMEOUT = 60
+
+
+@click.command()
+@click.argument(
+    "type_",
+    metavar="TYPE",
+    type=click.Choice(ledger_apis_registry.supported_ids),
+    #default=DEFAULT_LEDGER,
```
Comment (DavidMinarsch) [inline aea/cli/transfer.py:48]: No default please

File: aea/cli/transfer.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    "--settle-timeout", type=int, default=DEFAULT_SETTLE_TIMEOUT
+)
+@click.pass_context
+@check_aea_project
+def transfer(click_context, type_, address, amount, fee, yes, settle_timeout):
+    """Get the wealth associated with the private key."""
+    ctx = cast(Context, click_context.obj)
+    if not yes:
+        click.confirm(
+            f'Transfer from {type_} wallet to {address}, amount {amount} with fee {fee}. Do you want to continue?',
```
Comment (DavidMinarsch) [inline aea/cli/transfer.py:73]:
```suggestion
            f'You are about to transfer from {own_address} to {address} on ledger {type_} the amount {amount} with fee {fee}. Do you want to continue?', 
```

File: aea/cli/transfer.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    t = time.time()
+    while True:
+        if time.time() - t > timeout:
+            raise TimeoutError()
+        if LedgerApis.is_transaction_settled(identifier, tx_digest):
+            return
+        time.sleep(1)
+
+
+def do_transfer(ctx: Context, identifier: str, address: Address, amount: int, tx_fee: int) -> str:
```
Comment (DavidMinarsch) [inline aea/cli/transfer.py:94]: Start with `click.echo("Starting transfer ...")`

File: aea/cli/transfer.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    wallet = get_wallet_from_context(ctx)
+    source_address = wallet.addresses.get(identifier)
+    if not source_address:
+        raise click.ClickException(f"{identifier} wallet is not registered!")
+
+    balance = int(try_get_balance(ctx.agent_config, wallet, identifier))
+    if amount + tx_fee > balance:
+        raise click.ClickException(f"Balance is not enough: {balance}!")
```
Comment (DavidMinarsch) [inline aea/cli/transfer.py:102]:
```suggestion
        raise click.ClickException(f"Balance is not enough! Available={balance}, required={required_balance}!")
```

File: aea/cli/transfer.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    """Get the wealth associated with the private key."""
+    ctx = cast(Context, click_context.obj)
+    if not yes:
+        click.confirm(
+            f'Transfer from {type_} wallet to {address}, amount {amount} with fee {fee}. Do you want to continue?', 
+            abort=True
+        )
+    tx_digest = do_transfer(ctx, type_, address, amount, fee)
+    if not tx_digest:
+        raise click.ClickException(f'failed to send!: {tx_digest}')
```
Comment (DavidMinarsch) [inline aea/cli/transfer.py:78]: if there is no tx_digest, makes no sense to print it

File: aea/cli/transfer.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    click.echo(f"Transaction set. Waiting to be settled.")
+    wait_tx_settled(type_, tx_digest, timeout=settle_timeout)
+    click.echo(f"Transfer done. Sent {amount} with fee {fee} to {address}")
```
Comment (DavidMinarsch) [inline aea/cli/transfer.py:81]: The waiting should be optional. By default we should not wait and simply print the tx_digest, waiting can be specified with `--sync` flag.

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
LGTM, some suggested modifications

File: aea/cli/transfer.py (hunk 7 above)
Comment (solarw) [reply aea/cli/transfer.py:81]: done

File: aea/cli/transfer.py (hunk 6 above)
Comment (solarw) [reply aea/cli/transfer.py:78]: done

File: aea/cli/transfer.py (hunk 7 above)
Comment (solarw) [reply aea/cli/transfer.py:81]: done

File: aea/cli/transfer.py (hunk 5 above)
Comment (solarw) [reply aea/cli/transfer.py:102]: done

File: aea/cli/transfer.py (hunk 4 above)
Comment (solarw) [reply aea/cli/transfer.py:94]: done

File: aea/cli/transfer.py (hunk 2 above)
Comment (solarw) [reply aea/cli/transfer.py:48]: done

File: aea/cli/utils/package_utils.py (hunk 1 above)
Comment (solarw) [reply aea/cli/utils/package_utils.py:529]: done

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**